tests/server.js

```diff
@@ -155,6 +155,15 @@ var GuzzleServer = function(port, log) {
         var body = JSON.stringify(that.requests);
         res.writeHead(200, 'OK', { 'Content-Length': body.length });
         res.end(body);
+      } else if (req.url == '/guzzle-server/read-timeout') {
+        if (that.log) {
+          console.log('Sleeping');
+        }
+        res.writeHead(200, 'OK');
+        res.write("sleeping 60 seconds ...\n");
+        setTimeout(function () {
+          res.end("slept 60 seconds\n");
+        }, 60*1000);
```

Comment:
The server takes 60 seconds to send end of the response, and the client has a 1 second read timeout. So the client times out while reading the response.